client/client_test.go

```diff
@@ -1208,17 +1209,12 @@ func TestPublishAfterInitServerHasSnapshotKey(t *testing.T) {
 }
 
 func testPublishWithData(t *testing.T, rootType string, serverManagesSnapshot bool) {
-	// Temporary directory where test files will be created
-	tempBaseDir, err := ioutil.TempDir("/tmp", "notary-test-")
-	defer os.RemoveAll(tempBaseDir)
-	assert.NoError(t, err, "failed to create a temporary directory: %s", err)
-
-	gun := "docker.com/notary"
 	ts := fullTestServer(t)
 	defer ts.Close()
 
-	repo, _ := initializeRepo(t, rootType, tempBaseDir, gun, ts.URL,
+	repo, _ := initializeRepo(t, rootType, "docker.com/notary", ts.URL,
 		serverManagesSnapshot)
+	defer os.RemoveAll(repo.baseDir)
```

Comment:
There seems to be an unfinished comment on line 1227:

` // asserts that adding to the given roles results in the targets actually`